internal/pkg/util/fs/overlay/overlay_set_linux.go

```diff
@@ -80,9 +101,36 @@ func (s Set) performIndividualMounts() error {
 func (s Set) performFinalMount(rootFsDir string) error {
 	// Try to perform actual mount
 	options := s.options(rootFsDir)
-	sylog.Debugf("Mounting overlay with rootFsDir %q, options: %q", rootFsDir, options)
-	if err := syscall.Mount("overlay", rootFsDir, "overlay", syscall.MS_NODEV, options); err != nil {
-		return fmt.Errorf("failed to mount %s: %s", rootFsDir, err)
+	unprivOls, err := UnprivOverlaysSupported()
+	if err != nil {
+		return fmt.Errorf("error encountered while checking for unprivileged overlay support in kernel: %s", err)
+	}
+
+	if unprivOls {
+		sylog.Debugf("Mounting overlay (via syscall) with rootFsDir %q, options: %q", rootFsDir, options)
+		if err := syscall.Mount("overlay", rootFsDir, "overlay", syscall.MS_NODEV, options); err != nil {
+			return fmt.Errorf("failed to mount %s: %s", rootFsDir, err)
+		}
+	} else {
+		fuseOlFsCmd, err := bin.FindBin("fuse-overlayfs")
+		if err != nil {
+			return fmt.Errorf("kernel does not support unprivileged overlays, and fuse-overlayfs not installed: %s", err)
```

Comment:
```suggestion
			return fmt.Errorf("kernel does not support unprivileged overlays, and fuse-overlayfs not available: %w", err)
```

Reply:
It may actually be installed, but $PATH is set wrongly, or permissions are incorrect.